src/main/scala/com/microsoft/hyperspace/index/IndexLogEntryTags.scala

```diff
@@ -16,9 +16,50 @@
 
 package com.microsoft.hyperspace.index
 
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import com.microsoft.hyperspace.util.HyperspaceConf
+
 object IndexLogEntryTags {
-  // HYBRIDSCAN_REQUIRED indicates if Hybrid Scan is required for this index or not.
+  // HYBRIDSCAN_REQUIRED indicates if Hybrid Scan is required for the index or not.
   // This is set in getCandidateIndexes and utilized in transformPlanToUseIndex.
   val HYBRIDSCAN_REQUIRED: IndexLogEntryTag[Boolean] =
     IndexLogEntryTag[Boolean]("hybridScanRequired")
+
+  // IS_SIGNATURE_MATCH indicates if the plan has the same signature value with the index or not.
+  val IS_SIGNATURE_MATCH: IndexLogEntryTag[Boolean] =
+    IndexLogEntryTag[Boolean]("isSignatureMatch")
+
+  // IS_HYBRIDSCAN_CANDIDATE indicates if the index can be applied to the plan using Hybrid Scan.
+  val IS_HYBRIDSCAN_CANDIDATE: IndexLogEntryTag[Boolean] =
+    IndexLogEntryTag[Boolean]("hybridScanCandidate")
+
+  // IS_HYBRIDSCAN_CANDIDATE indicates if the index can be applied to the plan using Hybrid Scan.
+  val HYBRIDSCAN_CONFIG_CAPTURE: IndexLogEntryTag[Seq[String]] =
+    IndexLogEntryTag[Seq[String]]("hybridScanConfigCapture")
+
+  private def getHybridScanConfigs(spark: SparkSession): Seq[String] = {
+    Seq(
+      HyperspaceConf.hybridScanEnabled(spark).toString,
+      HyperspaceConf.hybridScanDeleteEnabled(spark).toString,
+      HyperspaceConf.hybridScanDeleteMaxNumFiles(spark).toString)
+  }
+
+  private[hyperspace] def resetHybridScanTagsIfNeeded(
+      spark: SparkSession,
+      plan: LogicalPlan,
+      indexes: Seq[IndexLogEntry]): Unit = {
+    val curConfigs = getHybridScanConfigs(spark)
+
+    indexes.foreach { index =>
+      val taggedConfigs = index.getTagValue(plan, HYBRIDSCAN_CONFIG_CAPTURE)
+      if (taggedConfigs.isEmpty || !taggedConfigs.get.equals(curConfigs)) {
```

Comment:
Question: It seems to be a force reset, so do we really need to check the previous value here? Why not skipping prev value check and simply doing an upsert and in case the prev value is the same as new value it is a simple overwrite of same value (which on avg should be O(1) - the same cost as prev value lookup).

Reply:
prev value check is required for `unsetTagValue(index.unsetTagValue(plan, IS_HYBRIDSCAN_CANDIDATE)) in any case. 
So I updated after the check.